packages/grafana-ui/src/components/FileUpload/FileUpload.tsx

```diff
@@ -42,37 +43,48 @@ export const FileUpload: FC<Props> = ({
 
   return (
     <>
-      <label className={cx(style.button, className)}>
+      <input
+        type="file"
+        id="fileUpload"
+        className={style.fileUpload}
+        onChange={onChange}
+        multiple={false}
+        accept={accept}
+        data-testid="fileUpload"
+      />
+      <label htmlFor="fileUpload" className={cx(style.labelWrapper, className)}>
         <Icon name="upload" className={style.icon} />
         {children}
-        <input
-          type="file"
-          id="fileUpload"
-          className={style.fileUpload}
-          onChange={onChange}
-          multiple={false}
-          accept={accept}
-        />
       </label>
+
       {fileName && (
-        <span aria-label="File name" className={style.fileName}>
+        <span aria-label="File name" className={style.fileName} data-testid="fileName">
           {trimFileName(fileName)}
         </span>
       )}
     </>
   );
 };
 
-const getStyles = stylesFactory((theme: GrafanaTheme2, size: ComponentSize) => {
+const getStyles = (size: ComponentSize) => (theme: GrafanaTheme2) => {
   const buttonStyles = getButtonStyles({ theme, variant: 'primary', size, iconOnly: false });
+  const focusStyle = getFocusStyles(theme);
+
   return {
-    fileUpload: css`
-      display: none;
-    `,
-    button: buttonStyles.button,
+    fileUpload: css({
```

Comment:
nit: can we align usage of the `css` function to use either the template literal or the object notation please?

Reply:
https://github.com/grafana/grafana/pull/47497/commits/dda51b02f66d4733409b5c0542c5faab3a2970c4 done